lib/rspec/rails/matchers/be_a_new.rb

```diff
@@ -49,7 +49,7 @@ def attributes
 
         def attributes_match?(actual)
           attributes.stringify_keys.all? do |key, value|
-            actual.attributes[key].eql?(value)
+            actual.attributes[key].eql?(value) || (value.respond_to?(:expected, true) && actual.attributes[key].include?(value.expected))
```

Comment:
Thanks! I'll dig in tomorrow and see where I get in that direction. I messed with it just now thinking it might just be that straight-forward...but then a bunch of tests blew up :/